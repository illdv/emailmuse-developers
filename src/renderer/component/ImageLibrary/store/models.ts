export interface IFileInfo {
  id: number;
  name: string;
  hash: string;
  imagable_id: number;
  imagable_type: string;
  created_at: string;
  updated_at: string;
  deleted_at?: string;
  url: string;
  thumb_url: string;
}

export interface IGetImagesResponse {
  current_page: number;
  data: IFileInfo[];
  first_page_url: string;
  from: number;
  last_page: number;
  last_page_url: string;
  next_page_url: null;
  path: string;
  per_page: number;
  prev_page_url: string;
  to: number;
  total: number;
}

export interface IImageLibraryItem {
  id: number;
  /**
   * Original image.
   */
  url: string;
  /**
   * Use for preview image.
   */
  thumb_url: string;
  name: string;
}