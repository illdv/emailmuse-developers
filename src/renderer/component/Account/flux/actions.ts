import { handleActions, createAction } from 'redux-actions';
import { Action } from 'redux';
import { IActionPayload } from 'src/renderer/flux/utils';

export interface IChangePasswordPayload {
  old_password: string;
  password: string;
  password_confirmation: string;
}
export interface IProfile{
  name: string;

}
const changePassword:IActionsChangePassword = {
  REQUEST: createAction('CHANGE_PASSWORD_REQUEST', (data:IChangePasswordPayload) => data ),
  SUCCESS: createAction('CHANGE_PASSWORD_SUCCESS'),
  FAILURE: createAction('CHANGE_PASSWORD_FAIL')
};
const getProfile = {
  REQUEST: createAction('GET_PROFILE_REQUEST'),
  SUCCESS: createAction('GET_PROFILE_SUCCESS', (data) => data),
  FAILURE: createAction('GET_PROFILE_FAIL', error => error)
};
interface IActionsChangePassword {
  REQUEST: (data?:IChangePasswordPayload) => IActionPayload<IChangePasswordPayload>;
  SUCCESS: () => IActionPayload<any>;
  FAILURE: (error) => IActionPayload<{error:string}>;
}
interface IActionsProfile {
  REQUEST: () =>  IActionPayload<any>;
  SUCCESS: ({name,email}:{name: string, email: string}) => IActionPayload<{name: string, email: string}>;
  FAILURE: () =>  IActionPayload<any>;
}
export namespace AccountSpace {
  interface IRequest {
    old_password: string;
    password: string;
    password_confirmation: string;
  }
  interface IActions {
    changePassword: IActionsChangePassword;
    getProfile: IActionsProfile;
  }
  export const Actions:IActions = {
    changePassword,
    getProfile
  };
}
