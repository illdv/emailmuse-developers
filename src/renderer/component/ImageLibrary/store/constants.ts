export const GET_IMAGES_REQUEST = 'GET_IMAGE_REQUEST';
export const GET_IMAGES_SUCCESS = 'GET_IMAGE_SUCCESS';
export const GET_IMAGES_FAILURE = 'GET_IMAGE_FAILURE';

export const UPLOAD_IMAGES_REQUEST = 'UPLOAD_IMAGE_REQUEST';
export const UPLOAD_IMAGES_SUCCESS = 'UPLOAD_IMAGE_SUCCESS';
export const UPLOAD_IMAGES_FAILURE = 'UPLOAD_IMAGE_FAILURE';

export const UPDATE_IMAGE_REQUEST = 'UPDATE_IMAGE_REQUEST';
export const UPDATE_IMAGE_SUCCESS = 'UPDATE_IMAGE_SUCCESS';
export const UPDATE_IMAGE_FAILURE = 'UPDATE_IMAGE_FAILURE';

export const DELETE_IMAGES_REQUEST = 'DELETE_IMAGE_REQUEST';
export const DELETE_IMAGES_SUCCESS = 'DELETE_IMAGE_SUCCESS';
export const DELETE_IMAGES_FAILURE = 'DELETE_IMAGE_FAILURE';