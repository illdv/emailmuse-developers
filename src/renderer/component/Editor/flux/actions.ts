import { createActionCreator } from 'src/renderer/flux/utils';
import { IAsyncAction2 } from 'src/renderer/flux/interface';
import { IEditEntity } from 'src/renderer/component/Editor/flux/interface';

const createAsyncAction = createActionCreator('EDITOR');

const edit         = createAsyncAction('EDIT');
const save         = createAsyncAction('SAVE');
const close        = createAsyncAction('CLOSE');
const remove       = createAsyncAction('REMOVE');
const saveAndClose = createAsyncAction('SAVE_AND_CLOSE');

export const EditorActions: IEditorActions = {
  edit,
  save,
  close,
  remove,
  saveAndClose,
};

export interface IEditorActions {
  edit: IAsyncAction2<{ editEntity: IEditEntity }>;
  save: IAsyncAction2<{ editEntity: IEditEntity }>;
  close: IAsyncAction2;
  remove: IAsyncAction2<{ editEntity: IEditEntity }>;
  saveAndClose: IAsyncAction2<{ editEntity: IEditEntity }>;
}
