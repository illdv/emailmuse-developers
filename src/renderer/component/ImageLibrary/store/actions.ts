import * as constants from './constants';
import { createAction } from 'redux-actions';
import { IGetImagesResponse } from 'src/renderer/component/ImageLibrary/store/models';
import * as Status from 'src/renderer/common/PreloaderLayout/Status/constants';

export const getImagesRequest = (page = 1, name: string = '') => ({
  type: constants.GET_IMAGES_REQUEST,
  payload: { page, name},
  meta: { status: Status.LOADING },
});

export const getImagesSuccess = (response: IGetImagesResponse) => ({
  type: constants.GET_IMAGES_SUCCESS,
  payload: response,
  meta: { status: Status.LOADED },
});

export const getImagesFailure = createAction(constants.GET_IMAGES_FAILURE, null, () => ({ status: Status.FAILED }));

export const uploadImagesRequest = (files: File | File[]) => ({
  type: constants.UPLOAD_IMAGES_REQUEST,
  payload: [].concat(files),
  meta: { status: Status.LOADING },
});

export const uploadImagesSuccess =
  createAction(constants.UPLOAD_IMAGES_SUCCESS, null, () => ({ status: Status.LOADED }));

export const uploadImagesFailure =
  createAction(constants.UPLOAD_IMAGES_FAILURE, null, () => ({ status: Status.FAILED }));

export const updateImageRequest = (payload: { imageId: number, name: string }) => ({
  type: constants.UPDATE_IMAGE_REQUEST,
  payload,
  meta: { status: Status.LOADING },
});

export const updateImageSuccess =
  createAction(constants.UPDATE_IMAGE_SUCCESS, null, () => ({ status: Status.LOADED }));

export const updateImageFailure =
  createAction(constants.UPDATE_IMAGE_FAILURE, null, () => ({ status: Status.FAILED }));

export const deleteImagesRequest = (ids: number | number[]) => ({
  type: constants.DELETE_IMAGES_REQUEST,
  payload: [].concat(ids),
  meta: { status: Status.LOADING },
});

export const deleteImagesSuccess =
  createAction(constants.DELETE_IMAGES_SUCCESS, null, () => ({ status: Status.LOADED }));

export const deleteImagesFailure =
  createAction(constants.DELETE_IMAGES_FAILURE, null, () => ({ status: Status.FAILED }));
