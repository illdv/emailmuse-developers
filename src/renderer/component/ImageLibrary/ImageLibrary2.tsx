import * as React from 'react';
import { IStyle } from 'type/materialUI';
import { Paper, withStyles } from '@material-ui/core/';
import { TypeBackground } from '@material-ui/core/styles/createPalette';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import Button from '@material-ui/core/Button';
import TablePagination from '@material-ui/core/TablePagination';
import block from 'bem-ts';

import {
  deleteImagesRequest,
  getImagesRequest,
  updateImageRequest,
  uploadImagesRequest,
} from 'src/renderer/component/ImageLibrary/store/actions';
import {
  getCurrentPageSelector,
  getImagesURLSelector,
  getLastPageSelector,
  getPerPageSelector,
  getTotalImages,
} from 'src/renderer/component/ImageLibrary/store/selectors';
import { IImageLibraryItem } from 'src/renderer/component/ImageLibrary/store/models';
import 'src/renderer/component/ImageLibrary/ImageLibrary.scss';
import { ImageLibraryDialog } from 'src/renderer/component/ImageLibrary/ImageLibraryDialog';
import { ImageLibraryList } from './ImageLibraryList';
import { DragAndDropTarget } from './DragAndDropTarget';
import { IPagination } from 'src/renderer/common/List/interface';

const b = block('image-library');

namespace ImageLibrarySpace {
  export interface IProps {
    classes?: any;
    actions?: {
      getImagesRequest: typeof getImagesRequest,
      uploadImagesRequest: typeof uploadImagesRequest,
      deleteImagesRequest: typeof deleteImagesRequest,
      updateImageRequest: typeof updateImageRequest,
    };
    items: IImageLibraryItem[];
    pagination: IPagination;
  }

  export interface IState {
    openDialog: boolean;
    chosenImage: IImageLibraryItem;
  }
}

const styles: IStyle = theme => ({
  root: {
    width: '100%',
    height: '100%',
    backgroundColor: theme.palette.background.paper,
  },
});

const mapStateToProps = state => ({
  // items: getImagesURLSelector(state),
  // pagination: {
  //   current_page: getCurrentPageSelector(state),
  //   total: getTotalImages(state),
  //   last_page: getLastPageSelector(state),
  //   per_page: getPerPageSelector(state),
  // },
});

const mapDispatchToProps = dispatch => ({
  // actions: bindActionCreators(
  //   { getImagesRequest, uploadImagesRequest, deleteImagesRequest, updateImageRequest },
  //   dispatch),
});

export class ImageLibrary2 extends React.Component<ImageLibrarySpace.IProps, ImageLibrarySpace.IState> {
  constructor(props) {
    super(props);
    this.state = { openDialog: false, chosenImage: null };
  }

  componentDidMount() {
    // this.props.actions.getImagesRequest();
  }

  onDropFile = item => {
    if (item && item.files) {
      // this.props.actions.uploadImagesRequest(item.files);
    }
  }

  onUploadFiles = e => {
    if (e.target.files) {
      let files = [];
      // tslint:disable-next-line
      for (let i = 0; i < e.target.files.length; i++) {
        files = files.concat(e.target.files[i]);
      }
      if (files.length) {
        // this.props.actions.uploadImagesRequest(files);
      }
    }
  }

  onOpenImageInfo = (item: IImageLibraryItem) => () => {
    this.setState({ openDialog: true, chosenImage: item });
  }

  closeDialog = () => {
    this.setState({ openDialog: false, chosenImage: null });
  }

  deleteItem = (item: IImageLibraryItem) => () => {
    // this.props.actions.deleteImagesRequest(item.id);
  }

  updateItem = (item: IImageLibraryItem, name) => {
    // this.props.actions.updateImageRequest({ imageId: item.id, name });
  }

  // TODO implement onProgress

  onChangePage = (e, page) => {
    // this.props.actions.getImagesRequest(page + 1);
  }

  // TODO: implement properly when there is capability to change rows per currentPage
  onChangeRowsPerPage = e => {
    // console.log('Change rows', e.target.value);
  }

  render() {
    const { classes, pagination } = this.props;
    return (
      <Paper elevation={4} >
        <div className={b()}>
          <TablePagination
            component='div'
            count={pagination.total || 0}
            rowsPerPage={pagination.per_page || 0}
            rowsPerPageOptions={[15]}
            page={pagination.current_page - 1}
            backIconButtonProps={{
              'aria-label': 'Previous Page',
            }}
            nextIconButtonProps={{
              'aria-label': 'Next Page',
            }}
            onChangePage={this.onChangePage}
            onChangeRowsPerPage={this.onChangeRowsPerPage}
          />
          <DragAndDropTarget
            onDrop={this.onDropFile}
            showOverlay={true}
            overlayMessage={'Drop files here to add them to your image library'}
          >
            <div className={b('container')}>
              {
                // this.state.chosenImage &&
                // <ImageLibraryDialog
                //   item={this.state.chosenImage}
                //   onDeleteItem={this.deleteItem}
                //   onUpdateItem={this.updateItem}
                //   onClose={this.closeDialog}
                // />
              }
              <ImageLibraryList
                items={this.props.items}
                onDelete={this.deleteItem}
                onSelect={this.onOpenImageInfo}
              />
            </div>
          </DragAndDropTarget>
          <div className={b('footer')}>
            <Button color='primary'>
              <label htmlFor='upload'>
                Upload images
              </label>
              <input
                id='upload'
                className={b('upload-input')}
                type='file'
                multiple
                onChange={this.onUploadFiles}
              />
            </Button>
          </div>
        </div>
      </Paper>
    );
  }
}

export default withStyles(styles)
(connect(mapStateToProps, mapDispatchToProps)
(ImageLibrary2 as any));
