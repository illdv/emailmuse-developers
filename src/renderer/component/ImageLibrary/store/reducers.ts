import { IFileInfo, IPagination }from 'src/renderer/component/ImageLibrary/store/models';
import { handleActions } from 'redux-actions';
import * as constants from 'src/renderer/component/ImageLibrary/store/constants';

export namespace ImageLibrary {
  export interface IState {
    all: IFileInfo[];
    pagination: IPagination;
  }

  const createDefaultState = (): IState => {
    return {
      all: [],
      pagination: {
        current_page: null,
        last_page: null,
        per_page: null,
        total: null
      },
    };
  };

  export const reducer = handleActions(
    {
      [constants.GET_IMAGES_SUCCESS]: (state: IState, action): IState => {
        return { ...state, all: action.payload.data,
          pagination: {
            current_page: action.payload.current_page,
            last_page: action.payload.last_page,
            per_page: action.payload.per_page,
            total: action.payload.total,
          }
        };
      },
    },
    createDefaultState()
  );
}