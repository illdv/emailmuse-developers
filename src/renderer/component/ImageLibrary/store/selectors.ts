import { createSelector } from 'reselect';
import { IFileInfo, IImageLibraryItem } from 'src/renderer/component/ImageLibrary/store/models';

const imagesSelector = (state):IFileInfo[] => state.images.all;

export const getImagesURLSelector = createSelector(
  imagesSelector,
  (state: IFileInfo[]): IImageLibraryItem[] =>
    state.map(file => {
      return {
        id: file.id,
        url: file.url,
        name: file.name
      };
    })
);
