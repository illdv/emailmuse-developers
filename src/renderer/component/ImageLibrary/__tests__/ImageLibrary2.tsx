import * as React from 'react';
import {mount, shallow} from 'enzyme';
import { ImageLibrary2 as ImageLibrary} from '../ImageLibrary2';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import {bindActionCreators} from 'redux';
import {deleteImagesRequest, getImagesRequest, updateImageRequest, uploadImagesRequest} from '../store/actions';

describe('>>>ImageLibrary', () => {
    const initialState = {
        items: [],
        pagination: {
            current_page: 1,
            last_page: 3,
            per_page: 15,
            total: 39,
        },
         actions: jest.fn(),
    };
    let store, props, imageLibrary;

    // //First way
    // const mockStore = configureStore();
    // beforeEach(() => {
    //     store = mockStore(initialState);
    //     imageLibrary = mount( <Provider store={store}><ImageLibrary2 /> </Provider> );
    // });

    // Second way
    beforeEach(() => {
        props = initialState;
        imageLibrary = shallow( <ImageLibrary {...props} > 0 </ImageLibrary> );
    });

    it('render ImageLibrary component', () => {
        expect(imageLibrary).toMatchSnapshot();
    });
});
