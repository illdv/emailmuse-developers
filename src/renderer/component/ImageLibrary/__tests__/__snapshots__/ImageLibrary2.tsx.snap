// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>>ImageLibrary render ImageLibrary component 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ImageLibrary2
    actions={[MockFunction]}
    items={Array []}
    pagination={
      Object {
        "current_page": 1,
        "last_page": 3,
        "per_page": 15,
        "total": 39,
      }
    }
  >
     0 
  </ImageLibrary2>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <div
        className="image-library"
      >
        <WithStyles(TablePagination)
          backIconButtonProps={
            Object {
              "aria-label": "Previous Page",
            }
          }
          component="div"
          count={39}
          nextIconButtonProps={
            Object {
              "aria-label": "Next Page",
            }
          }
          onChangePage={[Function]}
          onChangeRowsPerPage={[Function]}
          page={0}
          rowsPerPage={15}
          rowsPerPageOptions={
            Array [
              15,
            ]
          }
        />
        <DropTarget(DragAndDropTarget)
          onDrop={[Function]}
          overlayMessage="Drop files here to add them to your image library"
          showOverlay={true}
        >
          <div
            className="image-library__container"
          >
            <ImageLibraryList
              items={Array []}
              onDelete={[Function]}
              onSelect={[Function]}
            />
          </div>
        </DropTarget(DragAndDropTarget)>
        <div
          className="image-library__footer"
        >
          <WithStyles(Button)
            color="primary"
          >
            <label
              htmlFor="upload"
            >
              Upload images
            </label>
            <input
              className="image-library__upload-input"
              id="upload"
              multiple={true}
              onChange={[Function]}
              type="file"
            />
          </WithStyles(Button)>
        </div>
      </div>,
      "elevation": 4,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <WithStyles(TablePagination)
            backIconButtonProps={
              Object {
                "aria-label": "Previous Page",
              }
            }
            component="div"
            count={39}
            nextIconButtonProps={
              Object {
                "aria-label": "Next Page",
              }
            }
            onChangePage={[Function]}
            onChangeRowsPerPage={[Function]}
            page={0}
            rowsPerPage={15}
            rowsPerPageOptions={
              Array [
                15,
              ]
            }
          />,
          <DropTarget(DragAndDropTarget)
            onDrop={[Function]}
            overlayMessage="Drop files here to add them to your image library"
            showOverlay={true}
          >
            <div
              className="image-library__container"
            >
              <ImageLibraryList
                items={Array []}
                onDelete={[Function]}
                onSelect={[Function]}
              />
            </div>
          </DropTarget(DragAndDropTarget)>,
          <div
            className="image-library__footer"
          >
            <WithStyles(Button)
              color="primary"
            >
              <label
                htmlFor="upload"
              >
                Upload images
              </label>
              <input
                className="image-library__upload-input"
                id="upload"
                multiple={true}
                onChange={[Function]}
                type="file"
              />
            </WithStyles(Button)>
          </div>,
        ],
        "className": "image-library",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "backIconButtonProps": Object {
              "aria-label": "Previous Page",
            },
            "component": "div",
            "count": 39,
            "nextIconButtonProps": Object {
              "aria-label": "Next Page",
            },
            "onChangePage": [Function],
            "onChangeRowsPerPage": [Function],
            "page": 0,
            "rowsPerPage": 15,
            "rowsPerPageOptions": Array [
              15,
            ],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <div
              className="image-library__container"
            >
              <ImageLibraryList
                items={Array []}
                onDelete={[Function]}
                onSelect={[Function]}
              />
            </div>,
            "onDrop": [Function],
            "overlayMessage": "Drop files here to add them to your image library",
            "showOverlay": true,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <ImageLibraryList
                items={Array []}
                onDelete={[Function]}
                onSelect={[Function]}
              />,
              "className": "image-library__container",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "items": Array [],
                "onDelete": [Function],
                "onSelect": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <WithStyles(Button)
              color="primary"
            >
              <label
                htmlFor="upload"
              >
                Upload images
              </label>
              <input
                className="image-library__upload-input"
                id="upload"
                multiple={true}
                onChange={[Function]}
                type="file"
              />
            </WithStyles(Button)>,
            "className": "image-library__footer",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <label
                  htmlFor="upload"
                >
                  Upload images
                </label>,
                <input
                  className="image-library__upload-input"
                  id="upload"
                  multiple={true}
                  onChange={[Function]}
                  type="file"
                />,
              ],
              "color": "primary",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Upload images",
                  "htmlFor": "upload",
                },
                "ref": null,
                "rendered": "Upload images",
                "type": "label",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "image-library__upload-input",
                  "id": "upload",
                  "multiple": true,
                  "onChange": [Function],
                  "type": "file",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
            ],
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <div
          className="image-library"
        >
          <WithStyles(TablePagination)
            backIconButtonProps={
              Object {
                "aria-label": "Previous Page",
              }
            }
            component="div"
            count={39}
            nextIconButtonProps={
              Object {
                "aria-label": "Next Page",
              }
            }
            onChangePage={[Function]}
            onChangeRowsPerPage={[Function]}
            page={0}
            rowsPerPage={15}
            rowsPerPageOptions={
              Array [
                15,
              ]
            }
          />
          <DropTarget(DragAndDropTarget)
            onDrop={[Function]}
            overlayMessage="Drop files here to add them to your image library"
            showOverlay={true}
          >
            <div
              className="image-library__container"
            >
              <ImageLibraryList
                items={Array []}
                onDelete={[Function]}
                onSelect={[Function]}
              />
            </div>
          </DropTarget(DragAndDropTarget)>
          <div
            className="image-library__footer"
          >
            <WithStyles(Button)
              color="primary"
            >
              <label
                htmlFor="upload"
              >
                Upload images
              </label>
              <input
                className="image-library__upload-input"
                id="upload"
                multiple={true}
                onChange={[Function]}
                type="file"
              />
            </WithStyles(Button)>
          </div>
        </div>,
        "elevation": 4,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <WithStyles(TablePagination)
              backIconButtonProps={
                Object {
                  "aria-label": "Previous Page",
                }
              }
              component="div"
              count={39}
              nextIconButtonProps={
                Object {
                  "aria-label": "Next Page",
                }
              }
              onChangePage={[Function]}
              onChangeRowsPerPage={[Function]}
              page={0}
              rowsPerPage={15}
              rowsPerPageOptions={
                Array [
                  15,
                ]
              }
            />,
            <DropTarget(DragAndDropTarget)
              onDrop={[Function]}
              overlayMessage="Drop files here to add them to your image library"
              showOverlay={true}
            >
              <div
                className="image-library__container"
              >
                <ImageLibraryList
                  items={Array []}
                  onDelete={[Function]}
                  onSelect={[Function]}
                />
              </div>
            </DropTarget(DragAndDropTarget)>,
            <div
              className="image-library__footer"
            >
              <WithStyles(Button)
                color="primary"
              >
                <label
                  htmlFor="upload"
                >
                  Upload images
                </label>
                <input
                  className="image-library__upload-input"
                  id="upload"
                  multiple={true}
                  onChange={[Function]}
                  type="file"
                />
              </WithStyles(Button)>
            </div>,
          ],
          "className": "image-library",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "backIconButtonProps": Object {
                "aria-label": "Previous Page",
              },
              "component": "div",
              "count": 39,
              "nextIconButtonProps": Object {
                "aria-label": "Next Page",
              },
              "onChangePage": [Function],
              "onChangeRowsPerPage": [Function],
              "page": 0,
              "rowsPerPage": 15,
              "rowsPerPageOptions": Array [
                15,
              ],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <div
                className="image-library__container"
              >
                <ImageLibraryList
                  items={Array []}
                  onDelete={[Function]}
                  onSelect={[Function]}
                />
              </div>,
              "onDrop": [Function],
              "overlayMessage": "Drop files here to add them to your image library",
              "showOverlay": true,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <ImageLibraryList
                  items={Array []}
                  onDelete={[Function]}
                  onSelect={[Function]}
                />,
                "className": "image-library__container",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "items": Array [],
                  "onDelete": [Function],
                  "onSelect": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "div",
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <WithStyles(Button)
                color="primary"
              >
                <label
                  htmlFor="upload"
                >
                  Upload images
                </label>
                <input
                  className="image-library__upload-input"
                  id="upload"
                  multiple={true}
                  onChange={[Function]}
                  type="file"
                />
              </WithStyles(Button)>,
              "className": "image-library__footer",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <label
                    htmlFor="upload"
                  >
                    Upload images
                  </label>,
                  <input
                    className="image-library__upload-input"
                    id="upload"
                    multiple={true}
                    onChange={[Function]}
                    type="file"
                  />,
                ],
                "color": "primary",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Upload images",
                    "htmlFor": "upload",
                  },
                  "ref": null,
                  "rendered": "Upload images",
                  "type": "label",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "image-library__upload-input",
                    "id": "upload",
                    "multiple": true,
                    "onChange": [Function],
                    "type": "file",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "input",
                },
              ],
              "type": [Function],
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
